app/decorators/case_transition_decorator.rb

```diff
@@ -19,18 +19,24 @@ def event_and_detail
 
   private
 
-  def event
-    state_machine = object.case.state_machine
-    state_machine.event_name(object.event)
+  def event_desc
+    description_for_event = has_detail_description_for_event
+    if description_for_event.nil?
+      event
+    else
+      description_for_event 
+    end
```

Comment:
```suggestion
    description_for_event || event
```

Reply:
@Polsonby  I was thinking about this but was worrying you dislike || as I did in my previous implementation :)

Reply:
heh - sorry! I'm a nightmare, I must seem so inconsistent 😂 

Hopefully it makes sense that there's a kind of spectrum - say on the left hand end you have "no conditionals at all" (pure OO design ala Smalltalk) then in the middle somewhere you have "the least number of conditionals possible" to on the right hand side "lots and lots of conditionals" and very procedural code - generally speaking we're trying to move as far to the left on that spectrum as possible.

Really it comes from trying not to ask what type of object you're dealing with. Imagine you have a conditional where you check something to see what type of an object you have, and then send a different message depending which object it is. That's not good design. We'd rather inject a player of the role and send it a single message, and let the object work out how to respond to that message. 

Asking something whether it's an instance of nil is basically the same sort of thing, but it's a little further to the "left" on that hypothetical spectrum above - we just want to send messages to objects that play roles, but short of that we want to simplify the logic and know as little as we can about the objects we're dealing with. You can't send messages to nil, and you get nil back a LOT of the time. So lots of nil checks result and iturns out in this sort of app, there's a LOT of checking for nil going on so we try to hide it as much as possible. The above is one way. 

Another way you'll see a lot, while I'm thinking about it, is the safe navigation operator `&.` which basically wraps the message to the right in a check for nil

![image](https://user-images.githubusercontent.com/1161161/82375099-fb581300-9a17-11ea-8130-df33f14c1917.png)

We also really like the ternary operator for reducing 5 line if/else statements to 1 line :) 
In Rails apps things like chained method calls and reaching across collections of objects are really common, for good or ill.